docs/explanation/security/attack-surface.rst

```diff
@@ -0,0 +1,106 @@
+   * - 9283
+     - MGR (Dashboard)
+     - Ceph Dashboard HTTPS access
+     - Access should be restricted via firewall. Authentication is required.
+   * - 9128
+     - MGR (Prometheus)
+     - Prometheus metrics endpoint
+     - Restrict access to monitoring servers via firewall.
+   * - Internal/Local
+     - microcephd
+     - Local API socket for microceph CLI
+     - Access controlled by filesystem permissions on the socket file within the snap's data directory.
+   * - 7443
+     - microcephd
+     - Inter-node communication (if clustered)
+     - Uses TLS. Must be firewalled from external access, allowing only cluster members.
+   * - 22
+     - SSH
+     - Host OS access
+     - Standard SSH hardening practices (key auth, restricted access, firewall).
+   * - Other
+     - Other Services
+     - Potentially other services on host
+     - Audit all open ports on the host system.
+
+
+Network protocols and endpoints
+-------------------------------
+
+* Ceph Protocol (Messenger v1/v2): Used for all internal Ceph communication (MON, OSD, MGR, MDS).
+  Messenger v2 (default in newer Ceph versions) provides encryption capabilities for data in transit.  
+* Microcluster Protocol: Used for communication between microcephd instances in a multi-node cluster.
+  This communication is secured using TLS.  
+* Cephx Authentication: Primary mechanism for authenticating Ceph internal and client
+  communication. It provides mutual authentication.  
+* HTTP/HTTPS (RGW): Used for S3/Swift access via the RADOS Gateway. HTTPS with
+  strong TLS configuration is best practice.  
+* SSH: Used for accessing the host system to run microceph commands and perform
+  system maintenance.  
+* Local Socket API (microcephd): Communication between microceph CLI and microcephd
+  occurs over a Unix domain socket, protected by filesystem permissions.
+
+Data interfaces
+---------------
+
+* Block Devices and Filesystems: OSDs interact directly with underlying storage
+  (disks, partitions, or files configured via microceph disk add). The OSD processes
+  require elevated privileges, managed within the snap's confinement.  
+* dqlite Database Files: microcephd reads/writes configuration and state to dqlite
+  database files located within the snap's data directory (e.g., /var/snap/microceph/common/state/).
+  Access is controlled by filesystem permissions.  
+* CephFS Mounts: Clients mounting CephFS interact via the Ceph kernel module or FUSE,
+  requiring Cephx authentication.
+
+Management infrastructure
+-------------------------
+
+The primary management attack surface is the host, snap environment, and the microcephd service:
+
+* microceph CLI: Accessing this command usually requires sudo privileges on the host.
+  Compromising a host would allow an attacker to impact the Ceph cluster.  
+* microcephd: Compromising the microcephd process could allow manipulation of the cluster
```

Comment:
Use inline formatting here and other relevant places for better readability, e.g. to highlight commands, file paths, code, etc. Refer to [rST syntax reference](https://canonical-starter-pack.readthedocs-hosted.com/latest/reference/rst-syntax-reference/#inline-formatting).